--- PyQwt-3.8/iqt/_iqt.cpp.upgrade	2003-11-11 09:00:24.000000000 +0100
+++ PyQwt-3.8/iqt/_iqt.cpp	2004-08-25 07:22:30.927873423 +0200
@@ -88,7 +88,7 @@
     {"disable_hook", (PyCFunction)iqt_disable_hook, METH_NOARGS, 0},
     {"enable_hook",  (PyCFunction)iqt_enable_hook,  METH_NOARGS, 0},
     {"interacting",  (PyCFunction)iqt_interacting,  METH_NOARGS, 0},
-    {0, 0},
+    {0, 0, 0, 0},
 };
 
 extern "C" DL_EXPORT(void)
--- PyQwt-3.8/numpy/qwt_numarray.cpp.upgrade	2003-11-24 00:31:48.000000000 +0100
+++ PyQwt-3.8/numpy/qwt_numarray.cpp	2004-08-25 07:22:30.928873226 +0200
@@ -1,6 +1,6 @@
 // qwt_numarray.cpp: encapsulates all of PyQwt's calls to the numarray C-API.
 // 
-// Copyright (C) 2001-2003 Gerard Vermeulen
+// Copyright (C) 2001-2004 Gerard Vermeulen
 // Copyright (C) 2000 Mark Colclough
 //
 // This file is part of PyQwt
@@ -38,10 +38,10 @@
 #include <qwt_array.h>
 #undef NO_IMPORT // to force: void **PyQwt_Numarray_PyArray_API;
 #define PY_ARRAY_UNIQUE_SYMBOL PyQwt_Numarray_PyArray_API
-#include <numarray/libnumarray.h>
+#include <numarray/arrayobject.h>
 
 void qwt_import_libnumarray() {
-    import_libnumarray();
+    import_array();
 }
 
 int try_NumarrayArray_to_QwtArray(PyObject *in, QwtArray<double> &out)
@@ -50,8 +50,8 @@
         return 0;
 
     // FIXME: how costly is PyArray_ContiguousFromObject?
-    PyArrayObject *array = (PyArrayObject *)
-    PyArray_ContiguousFromObject(in, PyArray_DOUBLE, 1, 0);
+    PyArrayObject *array = (PyArrayObject *)PyArray_ContiguousFromObject(
+	in, PyArray_DOUBLE, 1, 0);
     
     if (!array) {
         PyErr_SetString(PyExc_RuntimeError,
--- PyQwt-3.8/pyqt_distutils/sysconfig.py.upgrade	2003-11-02 10:47:27.000000000 +0100
+++ PyQwt-3.8/pyqt_distutils/sysconfig.py	2004-08-25 07:22:30.929873028 +0200
@@ -64,7 +64,7 @@
                 '%(CXXFLAGS_WARN_ON)s '
                 ) % make_info
 
-        if make_info['RPATH']:
+        if make_info.get('RPATH'):
             linker_so = (
                 '%(CXX)s '
                 '%(RPATH)s%(LIBDIR_QT)s '
--- PyQwt-3.8/setup.py.upgrade	2003-11-25 07:24:53.000000000 +0100
+++ PyQwt-3.8/setup.py	2004-08-25 07:22:30.926873620 +0200
@@ -24,7 +24,7 @@
 name = 'PyQwt'
 qwtdir = 'qwt-sources'
 snapshot = '%04d%02d%02d' % (time.localtime()[:3])
-version = '3.8'
+version = '3.8b'
 
 #
 # CHECK FOR A RECENT PYTHON
--- PyQwt-3.8/Makefile.upgrade	2003-11-25 09:05:22.000000000 +0100
+++ PyQwt-3.8/Makefile	2004-08-25 07:22:30.924874015 +0200
@@ -60,7 +60,7 @@
 	if [ -e tmp/qwt ]; then \
 	    (cd tmp; cvs -q -d $(CVS-QWT) update -D $(CVS-DATE) -P -d qwt); \
 	else \
-	    (cd tmp; cvs -q -d $(CVS-QWT) checkout -D $(CVS-DATE) -P -d qwt); \
+	    (cd tmp; cvs -q -d $(CVS-QWT) checkout -D $(CVS-DATE) qwt); \
 	fi
 	cp -vpur tmp/qwt qwt-sources
 	find $(CVS-TABS) | xargs perl -pi -e 's|\t|    |g'
@@ -70,9 +70,9 @@
 	rm -rf qwt-sources
 	mkdir -p tmp
 	if [ -e tmp/qwt ]; then \
-	    (cd tmp; cvs -q -d $(CVS-QWT-SSH) update -P -d qwt); \
+	    (cd tmp; cvs -q -d $(CVS-QWT-SSH) update qwt); \
 	else \
-	    (cd tmp; cvs -q -d $(CVS-QWT-SSH) checkout -P -d qwt); \
+	    (cd tmp; cvs -q -d $(CVS-QWT-SSH) checkout -P qwt); \
 	fi
 	cp -vpur tmp/qwt qwt-sources
 	find $(CVS-TABS) | xargs perl -pi -e 's|\t|    |g'
