--- PyQwt-3.7/pyqt_distutils/configure.py.upgrade	Sun Jul  6 13:19:36 2003
+++ PyQwt-3.7/pyqt_distutils/configure.py	Sat Sep  6 16:14:50 2003
@@ -641,7 +641,7 @@
                 libraries.append(make['LIBS_THREAD'])
                 define_macros.append(('QT_THREAD_SUPPORT', None))
             else:
-                libraries.append(make_info['LIBS_QT'])
+                libraries.append(make['LIBS_QT'])
             # normalize, chop of the leading '-l'
             libraries = [ lib[2:] for lib in ' '.join(libraries).split()]
             
@@ -727,6 +727,8 @@
         '3.1.0': 'Qt_3_1_0',
         '3.1.1': 'Qt_3_1_1',
         '3.1.2': 'Qt_3_1_2',
+        '3.2.0': 'Qt_3_2_0',
+        '3.2.1': 'Qt_3_2_0',
         }.get(qt_version_str)
 
     if not qt:
@@ -772,7 +774,8 @@
             key = line[:split].strip()[6:]
             value = line[split+1:].strip()
             if value[:2] == "$$":
-                value = info[value[8:]]
+                macro = value.split()[0]
+                value = value.replace(macro, info[macro[8:]])
             info[key] = value.replace('$(QTDIR)', qtdir)
     return info
 
--- PyQwt-3.7/pyqt_distutils/sysconfig.py.upgrade	Sun Jul  6 12:55:41 2003
+++ PyQwt-3.7/pyqt_distutils/sysconfig.py	Wed Aug 20 23:32:08 2003
@@ -51,6 +51,7 @@
                 preprocessor = '%s %s' % (ccache, preprocessor)
             compiler_so = (
                 '%(CXX)s '
+                '%(CXXFLAGS)s '
                 '%(CXXFLAGS_THREAD)s '
                 '%(CXXFLAGS_RELEASE)s '
                 '%(CXXFLAGS_SHLIB)s '
@@ -68,6 +69,7 @@
                 ) % make_info
             compiler_so = (
                 '%(CXX)s '
+                '%(CXXFLAGS)s '
                 '%(CXXFLAGS_RELEASE)s '
                 '%(CXXFLAGS_SHLIB)s '
                 '%(CXXFLAGS_WARN_ON)s '
--- PyQwt-3.7/sip/qwtplot.sip.upgrade	Sun Jun 15 11:15:58 2003
+++ PyQwt-3.7/sip/qwtplot.sip	Sat Sep  6 16:21:00 2003
@@ -157,7 +157,7 @@
     int margin() const;
 
     // Title
-    void setTitle(const char *);
+    void setTitle(const QString &);
     QString title() const;
     void setTitleFont (const QFont &);
     QFont titleFont() const;
@@ -450,7 +450,7 @@
     QFont markerFont(long) const;
     bool setMarkerPen(long, const QPen &);
 
-    bool setMarkerLabel(long, const char *);
+    bool setMarkerLabel(long, const QString &);
     const QString& markerLabel(long) const;
     bool setMarkerLabelAlign(long, int);
     int markerLabelAlign(long) const;
